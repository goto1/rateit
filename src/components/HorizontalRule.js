import styled from "styled-components";
import PropTypes from "prop-types";

const HorizontalRule = styled.hr`
  border: 0px;
  height: 1px;
  width: ${props => props.width};
  margin: ${props => props.margin};
  background-image: linear-gradient(
    to left,
    ${props => props.colorOne},
    ${props => props.colorTwo},
    ${props => props.colorOne}
  );
`;

HorizontalRule.propTypes = {
  width: PropTypes.string.isRequired,
  margin: PropTypes.string.isRequired,
  colorOne: PropTypes.string.isRequired,
  colorTwo: PropTypes.string.isRequired
};

export default HorizontalRule;
