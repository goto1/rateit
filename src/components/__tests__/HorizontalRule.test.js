import React from "react";
import HorizontalRule from "../HorizontalRule";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";
import "jest-styled-components";

const props = {
  width: "90%",
  margin: "5px",
  colorOne: "#E5E5E5",
  colorTwo: "#0B7EFF"
};

const tree = renderer.create(<HorizontalRule {...props} />).toJSON();

describe("HorizontalRule component", () => {
  it("renders correctly", () => {
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  it("renders with specified styling rules", () => {
    expect(tree).toHaveStyleRule("width", "90%");
    expect(tree).toHaveStyleRule("margin", "5px");
  });
});
