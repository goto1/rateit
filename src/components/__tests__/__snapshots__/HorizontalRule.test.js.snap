// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalRule component renders correctly 1`] = `
.dwHqCE {
  border: 0px;
  height: 1px;
  width: 90%;
  margin: 5px;
  background-image: linear-gradient(    to left,    #E5E5E5,    #0B7EFF,    #E5E5E5  );
}

<hr
  className="sc-bdVaJa dwHqCE"
  width="90%"
/>
`;
